utils/markdown.ts

```diff
@@ -1,22 +1,61 @@
-const MARKDOWN_LINK_RE =
-  /(?<link>\[.*?]\((?<url>.*?)\)|<img.*?src="(?<url2>.*?)".*?>)/g;
+const HTML_LINK_RE = /(<img.*?src="(?<url>.*?)".*?>)/g;
+const HTML_LINK_NESTED_RE = /src="(?<url>.*?)"/g;
+
+const MARKDOWN_LINK_RE = /(\[.*?]\((?<url>.*?)\))/g;
+const MARKDOWN_LINK_NESTED_RE = /\((?<url>.*?)\)/g;
+
+function alreadyURL(path: string) {
+  return path.startsWith("http") || path.startsWith("https");
```

Comment:
refactor: starting with `http`, implies starting with `https` too

```suggestion
  return path.startsWith("http");
```